packages/gatsby/cache-dir/json-store.js

```diff
@@ -2,7 +2,18 @@ import React from "react"
 
 import PageRenderer from "./page-renderer"
 import { StaticQueryContext } from "gatsby"
-import socketIo, { getStaticQueryData, getPageQueryData } from "./socketIo"
+import {
+  getStaticQueryData,
+  getPageQueryData,
+  registerPath as socketRegisterPath,
+  unregisterPath as socketUnregisterPath,
+} from "./socketIo"
+
+if (process.env.NODE_ENV === `production`) {
+  throw new Error(
+    `It appears like Gatsby is misconfigured. JSONStore shouldn't be used in production.`
```

Comment:
I guess this _should_ never happen? Can the message be updated with advice on what to do if it does happen?

Reply:
I expanded this message in https://github.com/gatsbyjs/gatsby/pull/6777/commits/acfebf50fdeb3e360575d9e574f634e6e7d2d259 , but really only advice there is to report it as issue here